From e6438b9a9778abd81185237eea633a4348c8e183 Mon Sep 17 00:00:00 2001
From: Giovanni Mascellani <gmascellani@codeweavers.com>
Date: Tue, 22 Jul 2025 12:30:57 +0200
Subject: [PATCH] winepulse.drv: Allow 32-bit PCM audio samples.

They're already allowed on the ALSA and CoreAudio drivers.
---
 dlls/winepulse.drv/pulse.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/winepulse.drv/pulse.c b/dlls/winepulse.drv/pulse.c
index 4b287578e8b..9a218e75c1b 100644
--- a/dlls/winepulse.drv/pulse.c
+++ b/dlls/winepulse.drv/pulse.c
@@ -962,6 +962,8 @@ static HRESULT pulse_spec_from_waveformat(struct pulse_stream *stream, const WAV
             stream->ss.format = PA_SAMPLE_U8;
         else if (fmt->wBitsPerSample == 16)
             stream->ss.format = PA_SAMPLE_S16LE;
+        else if (fmt->wBitsPerSample == 32)
+            stream->ss.format = PA_SAMPLE_S32LE;
         else
             return AUDCLNT_E_UNSUPPORTED_FORMAT;
         pa_channel_map_init_auto(&stream->map, fmt->nChannels, PA_CHANNEL_MAP_ALSA);
-- 
GitLab

